SchedulerBot/SchedulerBot.Infrastructure.BotConnector/MessageProcessor.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using SchedulerBot.Business.Entities;
+using SchedulerBot.Database.Entities;
+using SchedulerBot.Infrastructure.Interfaces.BotConnector;
+
+namespace SchedulerBot.Infrastructure.BotConnector
+{
+	public class MessageProcessor : IMessageProcessor
+	{
+		private readonly AppCredentials appCredentials;
+		private readonly ILogger<MessageProcessor> logger;
+
+		public MessageProcessor(
+			AppCredentials appCredentials,
+			ILogger<MessageProcessor> logger)
+		{
+			this.appCredentials = appCredentials;
+			this.logger = logger;
+		}
+
+		public async Task SendMessageAsync(ScheduledMessage scheduledMessage, CancellationToken cancellationToken)
```

Comment:
Not sure it's significant in our case, but we could check if the cancellation token is cancelled in the beginning of the method, and if yes , then just do nothing.

Same applies to `ReplyAsync` if we include a cancellation token into its signature.